ipaserver/plugins/server.py

```diff
@@ -608,12 +608,12 @@ def _remove_server_principal_references(self, master):
             dn = DN(('cn', 'default'), ('ou', 'profile'), env.basedn)
             ret = conn.get_entry(dn)
             srvlist = ret.single_value.get('defaultServerList', '')
-            srvlist = srvlist[0].split()
+            srvlist = srvlist.split()
             if master in srvlist:
                 srvlist.remove(master)
                 attr = ' '.join(srvlist)
-                mod = [(ldap.MOD_REPLACE, 'defaultServerList', attr)]
-                conn.conn.modify_s(str(dn), mod)
+                ret['defaultServerList'] = attr
+                conn.update_entry(ret)
         except (errors.NotFound, ldap.NO_SUCH_ATTRIBUTE,
                 ldap.TYPE_OR_VALUE_EXISTS):
```

Comment:
since you now call `update_entry` instead of raw `modify_s`, you can replace `ldap.NO_SUCH_ATTRIBUTE` and `ldap.TYPE_OR_VALUE_EXISTS` exceptions with `errors.MidairCollision` and `errors.EmptyModlist` (see `error_handler` method of `LDAPClient` in `ipapython/ipaldap.py` for more info about raw ldap -> ipalib.errors mapping)